enterprise/cmd/frontend/internal/auth/oauth/middleware.go

```diff
@@ -56,24 +56,31 @@ func NewMiddleware(db database.DB, serviceType, authPrefix string, isAPIHandler
 
 		// If the actor is authenticated and not performing an OAuth flow, then proceed to
 		// next.
+		//
+		// If a sign-out cookie has been set during a previous sign-out request, remove it by setting MaxAge < 0.
 		if actor.FromContext(ctx).IsAuthenticated() {
+			if common.HasSignOutCookie(r) {
+				http.SetCookie(w, &http.Cookie{Name: common.SignoutCookie, Value: "", MaxAge: -1})
+			}
```

Comment:
See comment above: I'd move the cookie name and the `HassignoutCookie` method to non-enterprise code and then use it here.

I'd also move this logic here to that same file (that I mentioned in previous comment) so it will look kinda like this:

```suggestion
			auth.RemoveSignOutCookieIfSet(r) {
```

Under the hood this function then does that the code here previously did, except all the logic for a single cookie is now in one file (well, as much as possible anyway)